app/src/main/java/cm/aptoide/pt/abtesting/AbTestCacheValidator.java

```diff
@@ -0,0 +1,35 @@
+package cm.aptoide.pt.abtesting;
+
+import java.util.HashMap;
+
+public class AbTestCacheValidator {
+
+  private HashMap<String, ExperimentModel> localCache;
+
+  public AbTestCacheValidator(HashMap<String, ExperimentModel> localCache) {
+    this.localCache = localCache;
+  }
+
+  public boolean validateCache(String id) {
+    return localCache.containsKey(id) && !localCache.get(id)
+        .hasError() && !localCache.get(id)
+        .getExperiment()
+        .isExperimentOver();
+  }
+
+  public boolean validateExperiment(String id) {
+    return !localCache.get(id)
+        .getExperiment()
+        .isExpired() && !localCache.get(id)
+        .hasError() && !localCache.get(id)
+        .getExperiment()
+        .isExperimentOver() && localCache.get(id)
+        .getExperiment()
+        .isPartOfExperiment();
+  }
+
+  public void updateCache(String id, ExperimentModel experiment) {
+    if (localCache.containsKey(id)) localCache.remove(id);
```

Comment:
remove runs well even if there’s no association to the id, so the verification is not needed

Reply:
plus since this is a hashMap if you put something with the same id, it replaces the previous one, so this method could have only one line

Reply:
fixed